src/contents/en-gb/doc_dotnet.md

```diff
@@ -0,0 +1,326 @@
+
+## Read mutable data entries
+
+Now the mutable data is stored on the network with multiple entries, these can be retrieved using an `EntriesHandle`. To get `EntriesHandle` use `GetHandleAsync` function in `MDEntries` class and `MDInfo` of mutable data.
+
+Here is a method to fetch mutable data entries from network:
+```csharp
+using (var entriesHandle = await session.MDataEntries.GetHandleAsync(mDataInfo))
+{
+    var entries = await session.MData.ListEntriesAsync(entriesHandle)
+    foreach (var entry in entries)
+    {
+        var key = entry.Key.Val;
+        ver value = entry.Value.Content;
+    }
+}
+```
+This method gets a list of entries from the `ListEntriesAsync` function. We can iterate through the list to get the key and respective value for every entry.
+
+## Update and remove entries
+
+To update or remove entries create a mutation transaction (as above) with "update" and/or "remove" actions, and apply the mutations to the mutable data. `MDataEntryActionHanle` and `key` are required to perform these operations:
+
+**To update an entry:**
+```csharp
+var keys = await session.MData.ListKeysAsync(mdInfo);
+var keyToUpdate = keys[0];
+var newValue = "NewDataValue";
+using (var entriesHandle = await session.MDataEntryActions.NewAsync())
+{
+    var value = await session.MData.GetValueAsync(mdInfo, keyToUpdate.Val);
+    await session.MDataEntryActions.UpdateAsync(entriesHandle, keyToUpdate.Val, newValue.ToUtfBytes(), value.Item2 + 1);
+    await session.MData.MutateEntriesAsync(mdInfo, entriesHandle);
+}
+```
+
+**To remove an entry:**
+```csharp
+var keys = await session.MData.ListKeysAsync(mdInfo);
+var keyToDelete = keys[0];
+using (var entriesHandle = await session.MDataEntryActions.NewAsync())
+{
+    var value = await session.MData.GetValueAsync(mdInfo, keyToDelete.Val);
+    await session.MDataEntryActions.DeleteAsync(entriesHandle, keyToDelete.Val, value.Item2 + 1);
+    await session.MData.MutateEntriesAsync(mdInfo, entriesHandle);
+}
+```
+
+For both update and remove (not insert) a 1 is added to `value.Item2` and passed to the entry action function. This allows for versioning.
+
+## Versioning in mutable data
+
+Each entry in a mutable data has a version associated to it - a numeric value. 
+
+When a new entry is inserted it's inserted with version 0. Every time a mutation is performed this version increases by one, as specified in the above code. This is used by the network to ensure only one mutation is applied when simultaneous mutation requests are received for the same version of an entry. This ensures the state change of such an entry is what the originator of the request is intending to do.
+
+When an entry is removed it is never deleted from the mutable data, only the value is cleared. This ensures a new entry with the same key cannot be inserted - rather it is updated (ravi: does this sentence read correctly? I think I've understood what you meant but please double check). 
```

Comment:
Looks like there's a comment from your notes still in at the end of this sentence

Reply:
Will update the doc once the content is finalized on the HackMD.